test/integration/functional_test.go

```diff
@@ -70,9 +71,27 @@ var mitm *StartSession
 
 var runCorpProxy = detect.GithubActionRunner() && runtime.GOOS == "linux" && !arm64Platform()
 
+// k8s version string for functional test.
+// When this variable is set, minikube will be started with the specified version for functional test
+var k8sVersionString = ""
+
 // TestFunctional are functionality tests which can safely share a profile in parallel
 func TestFunctional(t *testing.T) {
+	testFunctional(t)
+}
 
+// TestFunctionalNewestKuberentes are functionality run functional tests using NewestKubernetesVersion
+func TestFunctionalNewestKuberentes(t *testing.T) {
+	if strings.Contains(*startArgs, "--kubernetes-version") {
+		t.Skip()
+	}
+	k8sVersionString = constants.NewestKubernetesVersion
```

Comment:
The problem is once we set this, we don't unset it, plus what about tests that run in parallel? The problem is `StartArgs()` is going to continue returning `--kubernetes-version=<newest>` for other tests that we don't want since this test is going to be ran in our full integration tests as well

Reply:
Indeed, it is my fault. What about putting this value into context? I think perhaps this can solve the problems you mentioned? Golang context is thread safe and does not affect other tests

Reply:
Revised. Now this value is put into context.